core-api/src/worker.rs

```diff
@@ -224,6 +225,28 @@ impl WorkerConfigBuilder {
             }
         }
 
+        if let Some(cache) = self.max_cached_workflows.as_ref() {
```

Comment:
Should we validate `max_outstanding_workflow_tasks` and `workflow_task_poller_behavior` are both at least 1 if workflow cache is disabled, or is this already asserted somewhere else?